crates/ironrdp-server/src/server.rs

```diff
@@ -229,37 +252,42 @@ impl RdpServer {
             acceptor.attach_static_channel(RdpsndServer::new(backend));
         }
 
+        let dcs_backend = DisplayControlBackend::new(Arc::clone(&self.display));
         let dvc = dvc::DrdynvcServer::new()
             .with_dynamic_channel(AInputHandler {
                 handler: Arc::clone(&self.handler),
             })
-            .with_dynamic_channel(DisplayControlServer);
+            .with_dynamic_channel(DisplayControlServer::new(Box::new(dcs_backend)));
         acceptor.attach_static_channel(dvc);
+    }
 
-        match ironrdp_acceptor::accept_begin(framed, &mut acceptor).await {
-            Ok(BeginResult::ShouldUpgrade(stream)) => {
+    pub async fn run_connection(&mut self, stream: TcpStream) -> Result<()> {
+        let framed = TokioFramed::new(stream);
+
+        let size = self.display.lock().await.size().await;
+        let capabilities = capabilities::capabilities(&self.opts, size);
+        let mut acceptor = Acceptor::new(self.opts.security.flag(), size, capabilities);
+
+        self.attach_channels(&mut acceptor);
+
+        let res = ironrdp_acceptor::accept_begin(framed, &mut acceptor)
+            .await
+            .context("Accept begin error")?;
```

Comment:
style: https://github.com/Devolutions/IronRDP/blob/ee30a72d279c8255b3e1b4bfd24d72d6603a3939/STYLE.md#formatting-of-error-messages